### DISCUSSION TITLE
Helper to get a list of `source-layer`s for a pmtiles file

### DISCUSSION THREAD

**Discussion by tordans** [protomaps/PMTiles#400] [Q&A]
I just had a wonderful debugging session related to the `source-layer` property of pmtiles/mbtiles and maplibre.
The issue is, that maplibre requires a specific `source-layer` to be set which must be present in the pmtiles.
However, there really is not good way to see the source layers that are available from inside my toolchain (maplibre, react map gl, maplibre pmtiles plugin).

Usually I work around this by setting the layer to `default` when I generate my pmtiles with tippecanoe. 

However this time, I forgot that this old dataset has a `feature.tippecanoe.layer` property which apparently overwrites the command line layer name.

I can inspect the layer names using the pm tiles preview in "vector_layers" "id".

My question is: Would it be possible to expose some tooling within the maplibre pmtiles package that allows to call "pmtiles.getLayerNames" which returns a list of layers? Or is there a different way I could use to give me some debugging help inside my application?

**Comment by bdon** [comment]
if you are using the source `url` field in the style there isn't really any point that the PMTiles object itself is exposed.

You could instantiate a PMTiles object and pass it in like this (more complicated)
https://github.com/protomaps/PMTiles/blob/main/js/examples/maplibre.html#L26

then call `getMetadata()` to get the JSON object and read the `vector_layers` there.
